hq_superset/views.py

```diff
@@ -140,26 +149,58 @@ def array_is_falsy(array_values):
 
     return array_values
 
+# ~5MB
+ASYNC_DATASOURCE_IMPORT_LIMIT_IN_BYTES = 5000000
+
+def trigger_datasource_refresh(domain, datasource_id, display_name):
+    if AsyncImportHelper(domain, datasource_id).is_import_in_progress():
+        flash("The datasource is already being imported in the background. "
+              "Please wait for it to finish before retrying",
+              "warning")
+        return redirect("/tablemodelview/list/")
+
+    provider = superset.appbuilder.sm.oauth_remotes["commcare"]
+    token = get_valid_cchq_oauth_token()
+    path, size = download_datasource(provider, token, domain, datasource_id)
+    datasource_defn = get_datasource_defn(provider, token, domain, datasource_id)
+    if size < ASYNC_DATASOURCE_IMPORT_LIMIT_IN_BYTES:
+        response = refresh_hq_datasource(
+            domain, datasource_id, display_name, path, datasource_defn, None
+            )
+        os.remove(path)
+        return response
+    else:
+        limit_in_mb = int(ASYNC_DATASOURCE_IMPORT_LIMIT_IN_BYTES / 1000000)
+        flash(f"The datasource is being refreshed in the background as it is"
+              " larger than {limit_in_mb} MB. This may take a while, please wait for it to finish",
+              "info")
```

Comment:
Totally open for dispute, but I feel like the `AsyncImportHelper` should encaptulate the `queue_refresh_task`, so that the view doesn't have to worry about this. I think it might be better to use the `AsyncImportHelper` as a "service" that handles all the bells and whistles around async tasks for the caller. Thoughts?

Reply:
If you look at `queue_refresh_task`, its just three lines, so there isn't a ton of code to move off of it to `AsyncImportHelper`. `AsyncImportHelper` is really just a group of related functions. I would hold off making abstractions until we really need to add a new task.